actions/build-push-to-dockerhub/action.yaml

```diff
@@ -25,17 +25,36 @@ inputs:
     description: |
       The dockerfile to use.
     required: false
+  build-args:
+    description: |
+      List of arguments necessary for the Docker image to be built.
+    required: false
+    default: ""
+  target:
+    description: |
+      Target stage to build
+    required: false
+  cache-from:
+    description: |
+      Where cache should be fetched from
+    required: false
+    default: "type=gha"
+  cache-to:
+    description: |
+      Where cache should be stored to
+    required: false
+    default: "type=gha,mode=max"
 
 runs:
   using: composite
   steps:
-    - name: Check if push is allowed
-      if: ${{ inputs.push == 'true' && github.event_name != 'push' }}
-      shell: sh
-      run: |
-        >&2 echo "Publishing to DockerHub is only allowed on push events."
-        >&2 echo "If you still want to build images without pushing them, set the push input to false."
-        exit 1
+    # - name: Check if push is allowed
+    #   if: ${{ inputs.push == 'true' && github.event_name != 'push' }}
```

Comment:
@Elfo404 why the logic here to only allow pushing on `push` events? I have a valid use case to push an image on a `workflow_dispatch` event.

Any thoughts on how to modify this in a non-breaking manner, or can I simply remove this logic?

Reply:
the original issue in https://github.com/grafana/shared-workflows/issues/27 mentioned limiting pushing to docker hub only for push events as a mechanism to prevent "non-prod" images to be pushed there. 

some more context here: https://github.com/grafana/shared-workflows/pull/26#discussion_r1444303741 between @zerok and @dsotirakis. as of "why" i'm not entirely sure, but i guess if that comment is still valid we could also allow jobs triggered by release_publish events to push to DH and it probably wouldn't be a breaking change.
 
opinions @dsotirakis?

Reply:
@joeyorlando my 2 cents here - I would say we can just say "allow all pushes that don't come from a pull request". the reason behind that is that we mostly want people to start using our GAR dev repositories for pushing from pull requests really. having said that, I can see the day that we will revisit this and we will start allowing pushes for **all** events, closer than ever 😁

Reply:
we can always add a small comment saying why we ideally wouldn't like to push to dockerhub from pull requests and what's the best alternative (push to GAR dev)